DOCS-7816-4.2-DW-types-intro-k (#1328)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-types.adoc

= Value Constructs for Types
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: studio, anypoint, esb, transform, transformer, format, aggregate, rename, split, filter convert, xml, json, csv, pojo, java object, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping

> DataWeave represents data using values, each of which has a data type associated with it. There are many types, such as strings, arrays, Booleans, numbers, objects, dates, times, and others. Each type supports several ways of creating its values. This topic explores many of the ways you can create them

Reviewer: ...just to avoid self-reference to the topic:
```suggestion
DataWeave represents data using values, each of which has a data type associated with it. There are many types, such as strings, arrays, Booleans, numbers, objects, dates, times, and others. Each type supports several ways of creating its values.
```

endif::[]
:keywords: studio, anypoint, esb, transform, transformer, format, aggregate, rename, split, filter convert, xml, json, csv, pojo, java object, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping

> DataWeave represents data using values where each value has a type associated. There are many types, such as strings, arrays, Booleans, numbers, objects, dates, and times among others. Each of these types has several ways of creating their values that we are going to explore in this topic.

Reviewer 2: ```suggestion
DataWeave represents data using values, each of which has a data type associated with it. There are many types, such as strings, arrays, Booleans, numbers, objects, dates, times, and others. Each type supports several ways of creating its values. This topic explores many of the ways you can create them.
```

Reviewer 2: want to make sure "representing" is ok to use there

Types in the `dw::Core` module (Core types) are available without having to import the Core module. Other modules need to be imported for their functions and types to be available.

> This topic describes the possible values for each DataWeave type, how these can be created, and also descriptions of common patterns, including:

Reviewer 2: ```suggestion
This topic describes the values that are possible for each DataWeave type and how to create those values. It also describes common patterns, including:
```

> This topic describes the values that are possible for each DataWeave type and how to create those values. It also describes common patterns, including:

Reviewer: To avoid self-reference, can we reword this? Maybe something like...:

It is important for you to know the values...type, how to create..., and the common patterns used:

[[dw_type_null]]
== Null (dw::Core Type)

DataWeave defines a Null data type for `null` values. It is important to note that `null`
is the value, while Null is the type. DataWeave functions do not recognize `Null` as
a `null` value.

You can determine whether a DataWeave function accepts `null` values by
looking in the DataWeave reference documentation for a function signature
that accepts a Null type. For example, the
xref:dw-core-functions-flatten.adoc#flatten2[flatten] function has two function
signatures, one of which is `flatten(Null): Null`. The signature indicates that
`flatten` can accept `null` values. The xref:dw-core-functions-plusplus.adoc[++]
(concatenate) function, which lacks a signature for handling value so the Null

> type, does not accept `null`.

Reviewer: The second sentence is not grammatical:

The function, which lacks...so the Null type, does not accept `null`?

Please clarify.
